docs(gateway/hybrid) add information regarding certificate revocation… (#2767)
Kong/docs.konghq.com: app/gateway-oss/2.4.x/hybrid-mode.md

`cluster_server_name` specifies the SNI (Server Name Indication extension) to use for data plane
connections to the control plane through TLS. When not set, data plane will use `kong_clustering` as the SNI.

## Revocation check of Data Plane certificates

> When Kong is running Hybrid mode with PKI support, the Control Plane can be configured to

Reviewer: we use the wording `PKI mode` above, should use the same description for matching (but having two "mode" in one sentence then feels verbose to me..)
```suggestion
When Kong is running Hybrid mode with PKI mode, the Control Plane can be configured to
```

optionally check for revocation status of the conecting Data Plane certificate.

> The supported method is through Online Certificate Status Protocol (OCSP) respoders.

Reviewer: ```suggestion
The supported method is through Online Certificate Status Protocol (OCSP) responders.
```

> In addition, Data Plane certificates issued must contain the "Certificate Authority Information Access" extension
> which points to the URI of which the OCSP responder may be reached from the Control Plane.

Reviewer: I feel many CAs won't let you touch raw individual extension so user may not aware about what `Certificate Authority Information Access` is. Maybe let's just say:
```suggestion
In addition, Data Plane certificates issued must contain the URI of OCSP responder may be reached from the Control Plane.
```
How do you think?

Reviewer 2: The suggested sentence is not grammatically correct. Suggest:

In addition, Data Plane certificates issued must contain the URI of OCSP responder that may be reached from the Control Plane.

Author: @fffonion @tlblessing thanks for the review.

My feeling is that in case the user manages their own internal CA, at least we can give some hint on what they should be looking for. So I will be proposing to combine those two:

> In addition, Data Plane certificates issued must contain the "Certificate Authority Information Access" extension that contains the URI of OCSP responder that may be reached from the Control Plane.

What do you folks think?

Reviewer 2: @dndx I think that's fine but I'd remove the double quotes.

Also, @javierguerragiraldez  is working on this same source file (different section) in PR https://github.com/Kong/docs.konghq.com/pull/2770#pullrequestreview-633713301 fyi.

Reviewer 2: Made your edit in commit 50f5747. There are 4 instances of "In addition", so I removed that as well. Even more refinements could be made here.

Author: Yes this looks good to me.

To enable OCSP checks, set the `cluster_ocsp` config on the Control Plane to one of the following values:

* `on`: OCSP revocation check is enabled and Data Plane must pass the revocation check in order to establish connection
with Control Plane. This implies that certificates without the OCSP extension or unreachable OCSP responder also prevents
connection from being established.
* `off`: OCSP revocation check is disabled (this is the default).

> * `optional`: OCSP revocation check will be attempted, however, if the required extension is not found inside Data Plane

Reviewer: If we go with the above comment, this becomes:
```suggestion
* `optional`: OCSP revocation check will be attempted, however, if the OCSP responder URI is not found inside Data Plane
```

provided certificate or communication with the OCSP responder failed, then Data Plane is still allowed through.

Note that OCSP checks are only performed on Control Plane against certificates provided by incoming Data Plane
nodes. The `cluster_ocsp` config has no effect on Data Plane nodes.

> 

Reviewer: As `cluster_oscp` affects telemetry as well and there's no `cluster_telemetry_ocsp`, and possibly generic messaing in the future, we could also mention:
```suggestion
`cluster_oscp` affects all Hybrid mode connections from Data Plane to Control Plane.

```
